tests/sanity/sanity_test.go

```diff
@@ -0,0 +1,356 @@
+
+	mockNodeService(fakeMounter, fakeMetadataService, mountPath)
+	mockControllerService(fakeCloud, mountPath)
+
+	driverOptions := &d.Options{
+		Mode:                              d.AllMode,
+		ModifyVolumeRequestHandlerTimeout: 60,
+		Endpoint:                          endpoint,
+	}
+
+	drv, err := d.NewDriver(fakeCloud, driverOptions, fakeMounter, fakeMetadataService, fakeKubernetesClient)
+	if err != nil {
+		t.Fatalf("Failed to create fake driver: %v", err.Error())
+	}
+	go func() {
+		if err := drv.Run(); err != nil {
+			panic(fmt.Sprintf("%v", err))
+		}
+	}()
+
+	config := csisanity.TestConfig{
+		TargetPath:           mountPath,
+		StagingPath:          stagePath,
+		Address:              endpoint,
+		DialOptions:          []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())},
+		IDGen:                csisanity.DefaultIDGenerator{},
+		TestVolumeSize:       10 * util.GiB,
+		TestVolumeAccessType: "mount",
+	}
+	csisanity.Test(t, config)
+}
+
+func createMockObjects(mockCtrl *gomock.Controller) (*mounter.MockMounter, *cloud.MockCloud, *metadata.MockMetadataService, *d.MockKubernetesClient) {
+	fakeMounter := mounter.NewMockMounter(mockCtrl)
+	fakeCloud := cloud.NewMockCloud(mockCtrl)
+	fakeMetadataService := metadata.NewMockMetadataService(mockCtrl)
+	fakeKubernetesClient := d.NewMockKubernetesClient(mockCtrl)
+
+	return fakeMounter, fakeCloud, fakeMetadataService, fakeKubernetesClient
+}
+
+func mockNodeService(m *mounter.MockMounter, md *metadata.MockMetadataService, mountPath string) {
+	m.EXPECT().Unpublish(gomock.Any()).Return(nil).AnyTimes()
+	m.EXPECT().GetDeviceNameFromMount(gomock.Any()).Return("", 0, nil).AnyTimes()
+	m.EXPECT().Unstage(gomock.Any()).Return(nil).AnyTimes()
+	m.EXPECT().FormatAndMountSensitiveWithFormatOptions(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+	m.EXPECT().NeedResize(gomock.Any(), gomock.Any()).Return(false, nil).AnyTimes()
+	m.EXPECT().MakeDir(gomock.Any()).DoAndReturn(
+		func(path string) error {
+			err := os.MkdirAll(path, os.FileMode(0755))
+			if err != nil {
+				if !os.IsExist(err) {
+					return err
+				}
+			}
+			return nil
+		},
+	).AnyTimes()
+	m.EXPECT().IsLikelyNotMountPoint(gomock.Any()).Return(true, nil).AnyTimes()
+	m.EXPECT().Mount(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+	m.EXPECT().FindDevicePath(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(
+		func(devicePath, volumeID, partition, region string) (string, error) {
+			if len(devicePath) == 0 {
+				return devicePath, cloud.ErrNotFound
+			}
+			return devicePath, nil
+		},
+	).AnyTimes()
+	m.EXPECT().PreparePublishTarget(gomock.Any()).Return(nil).AnyTimes()
+	m.EXPECT().IsBlockDevice(gomock.Any()).Return(true, nil).AnyTimes()
+	m.EXPECT().GetBlockSizeBytes(gomock.Any()).Return(int64(0), nil).AnyTimes()
```

Comment:
This seems to be the culprit that is causing the error `panic: reflect: Call using int32 as type int64`